osu.Game/Screens/Play/HUD/ArgonCounterTextComponent.cs

```diff
@@ -137,33 +133,48 @@ public ArgonCounterSpriteText(Func<char, string> getLookup)
             [BackgroundDependencyLoader]
             private void load(TextureStore textures)
             {
+                const string font_name = @"argon-counter";
+
                 Spacing = new Vector2(-2f, 0f);
-                Font = new FontUsage(@"argon-counter", 1);
-                glyphStore = new GlyphStore(textures, getLookup);
+                Font = new FontUsage(font_name, 1);
+                glyphStore = new GlyphStore(font_name, textures, getLookup);
             }
 
             protected override TextBuilder CreateTextBuilder(ITexturedGlyphLookupStore store) => base.CreateTextBuilder(glyphStore);
 
             private class GlyphStore : ITexturedGlyphLookupStore
             {
+                private readonly string fontName;
                 private readonly TextureStore textures;
                 private readonly Func<char, string> getLookup;
 
-                public GlyphStore(TextureStore textures, Func<char, string> getLookup)
+                private readonly Dictionary<char, ITexturedCharacterGlyph?> cache = new Dictionary<char, ITexturedCharacterGlyph?>();
+
+                public GlyphStore(string fontName, TextureStore textures, Func<char, string> getLookup)
                 {
+                    this.fontName = fontName;
                     this.textures = textures;
                     this.getLookup = getLookup;
                 }
 
                 public ITexturedCharacterGlyph? Get(string? fontName, char character)
                 {
+                    // We only service one font.
+                    Debug.Assert(fontName == this.fontName);
```

Comment:
Why does this assert but `LegacySpriteText` returns null? Commit message mentions "fallback weirdness" but I don't know what that means.